src/test/java/org/jabref/model/database/event/UpdateTimestampListenerTest.java

```diff
@@ -0,0 +1,99 @@
+package org.jabref.model.database.event;
+
+import org.jabref.gui.BasePanel;
+import org.jabref.logic.preferences.TimestampPreferences;
+import org.jabref.model.database.BibDatabase;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.preferences.JabRefPreferences;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class UpdateTimestampListenerTest {
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    private BibDatabase database;
+
+    @Before
+    public void setUp(){
+        database = new BibDatabase();
+    }
+
+    @Test
+    public void updateTimestampEnabled(){
+        final String TIMESTAMP_FIELD = "timestamp";
+        final String BASE_DATE = "2000-1-1";
+        final String NEW_DATE = "2000-1-2";
+        final boolean INCLUDE_TIMESTAMP = true;
+
+        JabRefPreferences preferencesMock = mock(JabRefPreferences.class);
```

Comment:
You have a few set-up steps that are shared between the tests methods. Could you please move them to the `setup` method.

Reply:
In my opinion, it is not necessary (or even useful) to declare these values beforehand.

The additional time spent when initializing three strings twice instead of only once should be negligible when running the tests. Furthermore, when looking at the test cases, defining the constants in the beginning of each test makes it much clearer exactly which values are used.

But that is just my opinion. What do you (or other contributors) think about this? If you still prefer that the constants are only initialized once, I will change it.
But, as they are constants, initializing them in the set-up would not be the best option in my mind. I would rather make them private final instance variables.

Reply:
I didn't meant the constants. These are clearly related to the specific test and thus can stay there. However, you have some common initialization (preferences, database with entry, ...) which should be extracted to `setup`. Not so much for performance but to make the tests smaller and thus easier to understand and to maintain.
Have look at for example https://github.com/JabRef/jabref/blob/effe3349e2dfe8b104ade137f86ef8e2fbee36a5/src/test/java/org/jabref/logic/cleanup/MoveFilesCleanupTest.java#L35-L54

Reply:
Thank you for clarifying. I get your point, but I am still not sure what you want me to move into the setup method.

I could move:
- the instantiation of the mocks for JabRefPreferences and TimeStampPreferences
- the setup to catch calls to JabRefPreferences#getTimestampPreferences()
- the instantiation of the BibEntry as well as its insertion into the database

This would also mean that the mocks and the BibEntry would have to be instance variables.

Would this meet your expectations?

If I should also move more of the initialization into the setup method, I don't see a way around making the constants "TIMESTAMP_FIELD", "BASE_DATE", and "NEW_DATE" instance variables.

Reply:
Yes, I meant exactly those three points! Thanks.

Reply:
@Siedlerchr This is the discussion about moving parts of the initialization into the setup method.